web/source/text/engineDeviceSpec.ts

```diff
@@ -0,0 +1,76 @@
+namespace com.keyman.text {
+  export enum Browser {
+    Chrome = 'chrome',
+    Edge = 'edge',
+    Firefox = 'firefox',
+    Native = 'native', // Used by embedded mode
+    Opera = 'opera',
+    Safari = 'safari',
+    Other = 'other'
+  }
+
+  export enum OperatingSystem {
+    Windows = 'Windows',
+    macOS = 'MacOSX',
+    Linux = 'Linux',
+    Android = 'Android',
+    iOS = 'iOS',
+    Other = 'other'
+  }
+
+  export enum FormFactor {
+    Desktop = 'desktop',
+    Phone = 'phone',
+    Tablet = 'tablet'
+  }
+
+  /**
+   * This class provides an abstract version of com.keyman.Device that is headless-friendly, 
+   * containing only the information needed by web-core for text processing use.
+   */
+  export class EngineDeviceSpec {
+    readonly browser: Browser;
+    readonly formFactor: FormFactor;
+    readonly OS: OperatingSystem;
+    readonly touchable: boolean;
+
+    constructor(browser: string, formFactor: string, OS: string, touchable: boolean) {
+      switch(browser as Browser) {
```

Comment:
Should we add `toLowerCase()` as you have done for the other switches below? I see you do a casting in the switch statement to `Browser` but you don't do the same for `formFactor` or `OS` -- why? See below for further thoughts.

Reply:
Ah, right.  Just pushed a commit for this.